views/layouts/MainLayout.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useEffect, useContext } from 'react'
+import { Outlet, useLocation, useNavigate } from 'react-router-dom'
+
+import Header from '../components/Header'
+import Sidebar from '../components/Sidebar'
+
+import getAvatar from '../fe-utils/avatarUtil'
+import getCounts from '../fe-utils/countUtil'
+import { fetchSubjects } from '../fe-utils/fetchUtil'
+import { AuthContext } from '../contexts/AuthContext'
+import { headerTitle } from './helpers'
+
+const MainLayout = ({ classes, theme }) => {
+  const [user] = useContext(AuthContext)
+  const { pathname } = useLocation()
+  const navigate = useNavigate()
+
+  const [openDrawer, setOpenDrawer] = useState(false)
+  const [sideBarState, setSideBarState] = useState({
+    totalDays: 0,
+    totalStudies: 0,
+    totalSubjects: 0,
+  })
+  const [avatar, setAvatar] = useState('')
+
+  const toggleDrawer = () => setOpenDrawer(!openDrawer)
+
+  useEffect(() => {
+    fetchSubjects().then((acl) => {
+      setSideBarState(getCounts({ acl }))
+    })
+    setAvatar(getAvatar({ user }))
+  }, [])
+
+  return (
+    <div className={classes.root}>
+      <Header
+        handleDrawerToggle={toggleDrawer}
+        title={headerTitle(pathname)}
+        isAccountPage={false}
+        classes={classes}
+      />
+      <Sidebar
+        avatar={avatar}
+        handleDrawerToggle={toggleDrawer}
+        mobileOpen={openDrawer}
+        totalDays={sideBarState.totalDays}
+        totalStudies={sideBarState.totalStudies}
+        totalSubjects={sideBarState.totalSubjects}
+        user={user}
+        classes={classes}
+        theme={theme}
```

Comment:
the sidebar is a functional component so we can get this data from context right?

Reply:
[4f93dab](https://github.com/AMP-SCZ/dpdash/pull/317/commits/4f93dabcfecae6eae748b2bcbb270933bc21546e)

Reply:
[8ce38fe](https://github.com/AMP-SCZ/dpdash/pull/317/commits/8ce38fe8347636204db05592d3d398ca7e87a361)